scripts/get-historical-lockers.ts

```diff
@@ -0,0 +1,207 @@
+import fs from 'fs';
+import { createObjectCsvWriter } from 'csv-writer';
+
+const hre = require("hardhat");
+const { ethers, network } = hre;
+const { provider, getContractAt, BigNumber} = ethers;
+
+
+/*
+    Penalty Rebate METHODOLOGY
+
+    To calculate a 50% penalty-free unlock, all we do
+    is divide the penalty amount in the current mechanism by two
+    and send this back to the user
+
+    We divide by two because 50% unlock corresponds to:
+    how much was kept originally + 50% of the penalty
+
+    Example (without 50% penalty-free unlock):
+
+    500 RBN, 2yr lock (75% penalty fee)
+    Keep: 500 RBN * 25% = 125 RBN
+    Penalty: 500 RBN * 75% = 375 RBN
+
+    Example (with 50% penalty-free unlock):
+
+    500 RBN, 2yr lock (75% penalty fee)
+    Keep: 500 RBN * 50% (penalty-free) + 250 RBN * 25% = 312.5 RBN
+    Penalty: 250 RBN * 75% = 187.5 RBN
+
+    So, user calls force_withdraw on 500 RBN
+    Initially, he keeps 125 RBN and 375 RBN is sent to penalty contract
+    Penalty contract divides 375 RBN by 2 = 187.5 RBN and sends it back to locker
+    Keep Final: 125 RBN + 187.5 RBN = 312.5 RBN
+    Penalty Final: 187.5 RBN = 187.5 RBN
+
+    This generalizes across all RBN lock sizes (not just 500 RBN) + lock times (not just 2yr)
+*/
+
+// ABI
+const ABI = [
+  "event Deposit(address indexed deposit_from, address indexed provider, uint256 value, uint256 indexed locktime, int128 type, uint256 ts)",
+];
+
+// Fork mainnet at block no 16699831
+const FORK_BLOCK_NUMBER = 16699831;
+// Use equivalent unix timestmap for block num
+// Date of RGP-31 passing
+// Fri Feb 24 2023 18:00:00 GMT+0000
+const FORK_BLOCK_TIMESTAMP = 1677261600;
+
+// This is for passing values into Penalty Escrow Contract
+const FILENAME_JSON = "addresses_penalties_rebate.json"
+// This is for users to see difference
+const FILENAME_CSV = "addresses_penalties_rebate.csv"
+
+const VE_RBN = "0x19854C9A5fFa8116f48f984bDF946fB9CEa9B5f7"
```

Comment:
This token from this address has no holders - maybe it's how it's supposed to be. just mentioning